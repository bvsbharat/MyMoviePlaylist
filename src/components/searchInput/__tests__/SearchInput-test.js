import mount from "@test";
import renderer from "react-test-renderer";
import SearchInput from "../SearchInput";

describe(SearchInput, () => {
    const component = mount(<SearchInput />);

    it("renders without crashing", () => {
        expect(component).toBeTruthy();
        const componentRender = renderer.create(
            <SearchInput
                setSearchValue={() => {}}
                placeholder="Search"
                value=""
            />,
        );

        const tree = componentRender.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
