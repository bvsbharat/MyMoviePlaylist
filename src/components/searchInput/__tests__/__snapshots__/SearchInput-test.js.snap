// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders without crashing 1`] = `
<div
  className="filterbox__searchbox"
>
  <div
    className="relative bg-deep-purple-accent-400"
  >
    <div
      className="absolute inset-x-0 bottom-0"
    >
      <svg
        className="w-full -mb-1 text-white"
        fill="currentColor"
        preserveAspectRatio="none"
        viewBox="0 0 224 12"
      >
        <path
          d="M0,0 C48.8902582,6.27314026 86.2235915,9.40971039 112,9.40971039 C137.776408,9.40971039 175.109742,6.27314026 224,0 L224,12.0441132 L0,12.0441132 L0,0 Z"
        />
      </svg>
    </div>
    <div
      className="px-4 py-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl md:px-24 lg:px-8 lg:py-20"
    >
      <div
        className="relative max-w-2xl sm:mx-auto sm:max-w-xl md:max-w-2xl sm:text-center"
      >
        <h2
          className="mb-6 font-sans text-3xl font-bold tracking-tight text-white sm:text-4xl sm:leading-none"
        >
          Search for your favorite movies
          <br
            className="hidden md:block"
          />
          to add in your
           
          <span
            className="relative inline-block px-2"
          >
            <div
              className="absolute inset-0 transform -skew-x-12 bg-teal-accent-400"
            />
            <span
              className="relative text-teal-900"
            >
              playlist
            </span>
          </span>
        </h2>
        <form
          className="flex flex-col items-center w-full mb-4 md:flex-row md:px-16"
          onSubmit={[Function]}
        >
          <input
            autoComplete="off"
            className="flex-grow w-full h-12 px-4 mb-3 text-white transition duration-200 border-2 border-transparent rounded appearance-none md:mr-2 md:mb-0 bg-deep-purple-900 focus:border-teal-accent-700 focus:outline-none focus:shadow-outline"
            name="movieName"
            onChange={[Function]}
            placeholder="Enter movie name here"
            required={true}
            type="text"
            value=""
          />
        </form>
        <p
          className="max-w-md mb-10 text-xs tracking-wide text-indigo-100 sm:text-sm sm:mx-auto md:mb-16"
        >
          Click on the arrow to scroll to search results.
        </p>
        <a
          aria-label="Scroll down"
          className="flex items-center justify-center w-10 h-10 mx-auto text-white duration-300 transform border border-gray-400 rounded-full hover:text-teal-accent-400 hover:border-teal-accent-400 hover:shadow hover:scale-110"
          href="#result"
        >
          <svg
            fill="currentColor"
            height="12"
            viewBox="0 0 12 12"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10.293,3.293,6,7.586,1.707,3.293A1,1,0,0,0,.293,4.707l5,5a1,1,0,0,0,1.414,0l5-5a1,1,0,1,0-1.414-1.414Z"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;
